### PR TITLE
Upgrade to TS 1.8

### LINKED ISSUE #596: Update TypeScript to 1.8 and update TypeDoc
#### Context

Typedoc has been updated to support 1.8.10 which means we can upgrade Excalibur to 1.8.

https://github.com/TypeStrong/typedoc/pull/204

#### Proposal

- [x] Update Excalibur to 1.8 and fix any compiler issues, run tests, verify
- [x] Update Excalibur docs repository to newest Typedoc and test docs still build properly on **current** (1.6) and **new** (1.8) versions of Ex
- [ ] Update site repo first
- [ ] Merge PR

### PR DESCRIPTION
**Pull request by kamranayub**
Closes #596 

Proposed Changes:
- Update to TypeScript 1.8
- Fix 1.8 compiler errors

DO NOT MERGE until docs site and Typedoc have been completely upgraded and are ready to roll.

### CODE REVIEW

**Comment by kamranayub** [comment]
'tis ready! :+1:

File: src/engine/Actor.ts [hunk 1]
```diff
@@ -848,7 +848,6 @@ module ex {
              return ex.Side.Top;
           }
        }
-       return ex.Side.None;
```
Comment (eonarheim) [inline src/engine/Actor.ts]: +1 for unreachable code removal

File: src/engine/Input/Gamepad.ts [hunk 2]
```diff
@@ -260,33 +260,34 @@
             this.at(i).navigatorGamepad = gamepads[i];
 
             // Buttons
-            var b: string, a: string, value: number, buttonIndex: number, axesIndex: number;
+            var b: string, bi: string|number, a: string, ai: string|number, value: number;
```
Comment (eonarheim) [inline src/engine/Input/Gamepad.ts]: What was the impetus for shortening the variable names to `bi` and `ai`?

File: src/engine/Input/Gamepad.ts [hunk 3]
```diff
@@ -260,33 +260,34 @@
...
+                           this.at(i).updateButton(bi, 0);
+                        }
                      }
                   }
                }
             }
 
             // Axes
             for (a in Axes) {
-               if (typeof Axes[a] !== 'number') { continue; }
```
Comment (eonarheim) [inline src/engine/Input/Gamepad.ts]: I do like the removal of continue statements, they break control flow in a weird way

**Pull request merged**